File: apps/customer-web/lib/cart-totals.test.ts
import { describe, expect, it } from "vitest"

import type { GuestCartLine } from "./cart-store"
import { computeCartTotals, formatCents, parseCents } from "./cart-totals"

const line = (unit: string, qty: number): GuestCartLine => ({
  productId: unit + qty,
  name: "x",
  imageUrl: null,
  unitPriceAmount: unit,
  currency: "AUD",
  quantity: qty,
  packageKey: "pkg_a",
})

describe("cents helpers", () => {
  it("round-trips", () => {
    expect(parseCents("5.00")).toBe(500)
    expect(parseCents("12.5")).toBe(1250)
    expect(formatCents(500)).toBe("5.00")
    expect(formatCents(99)).toBe("0.99")
  })
})

describe("computeCartTotals", () => {
  it("sums the item lines only — 021 drops the flat delivery fee", () => {
    const totals = computeCartTotals([line("5.00", 2), line("3.00", 1)])
    expect(totals.itemSubtotal).toBe("13.00")
    // Delivery is quoted per package at checkout — the cart never carries a fee (FR-024).
    expect(totals).not.toHaveProperty("deliveryFee")
    expect(totals).not.toHaveProperty("grandTotal")
  })

  it("is zero for an empty cart", () => {
    expect(computeCartTotals([]).itemSubtotal).toBe("0.00")
  })
})

File: apps/customer-mobile/shared/src/commonTest/kotlin/com/effyshopping/customer/mobile/features/checkout/CartTotalsTest.kt
package com.effyshopping.customer.mobile.features.checkout

import com.effyshopping.customer.mobile.features.cart.domain.GuestCartLine
import com.effyshopping.customer.mobile.features.cart.domain.computeTotals
import com.effyshopping.customer.mobile.features.cart.domain.formatCents
import com.effyshopping.customer.mobile.features.cart.domain.parseCents
import kotlin.test.Test
import kotlin.test.assertEquals

class CartTotalsTest {

    private fun line(unit: String, qty: Int) =
        GuestCartLine("p$unit$qty", "x", null, unit, "AUD", qty)

    @Test
    fun centsRoundTrip() {
        assertEquals(500L, parseCents("5.00"))
        assertEquals(1250L, parseCents("12.5"))
        assertEquals("5.00", formatCents(500))
        assertEquals("0.99", formatCents(99))
    }

    @Test
    fun sumsLinesItemSubtotalOnly() {
        // 021: the flat delivery fee is gone (FR-024) — the cart shows the item subtotal only.
        val totals = computeTotals(listOf(line("5.00", 2), line("3.00", 1)))
        assertEquals("13.00", totals.itemSubtotal)
    }

    @Test
    fun emptyCartHasZeroSubtotal() {
        assertEquals("0.00", computeTotals(emptyList()).itemSubtotal)
    }
}
